### DISCUSSION TITLE
AppImage flexibility

### DISCUSSION THREAD

**Discussion by mgord9518** [ivan-hc/AM#15] [Ideas]
Hello, I've done some quick skimming over the source code and it appears that this always uses an installation script even for AppImages? If you'd like, I have a project I've been working on for a while that contains some code snippets that could possibly help with installing AppImages without needing individual scripts.

The project is [here](https://github.com/mgord9518/shappimage/blob/main/runtime.sh) (it's intended to be an AppImage runtime implemented in shell script), and it contains functions to find SquashFS offset and parse (some parts) of ELF headers. With the help of squashfuse, all needed AppImage information could be extracted from the bundle itself and all AppImages could be installed needing nothing but a download URL.

**Comment by DLopezJr** [comment]
Are there any cons to this method?

**Comment by ivan-hc** [comment]
Sorry for the late reply, I'll answer you right now.

in fact it seems that the scripts are the same, but this is not always the case. Sometimes for AppImage installation scripts there are additional actions:
- to find the latest version, some of them require additional parameters and variables because they are "hidden" in a proprietary server with a certain type of encryption (about 200 AppImages on this site, including the "Inkskape", have a particularly different link;
- about 10% of the AppImages listed have their own integrated update system (using "zsync" and "appimageupdatetool"), for example the various "wine" and "libreoffice", the remaining 90% use a version comparison method at the source and the one installed, and it is not always the same (over 1500 AppImages come from Github, the others do not);
- some scripts have dedicated steps, for example instead of the symbolic link in $PATH they use a script to adapt the font size and use the system theme (see "Kdenlive");
- some scripts (for now only 2, Kdenlive and Inkscape) have a dedicated script to "roll back" to the previous version, and this is because they are not hosted on Github;
- AM not only manages AppImage, but also official portable applications such as Firefox, Blender, Visual Studio Code, Thunderbird, SuperTuxKart, the Platform Tools suite (ADB, fastboot...), NodeJS, Ventoy, Brave... a script like mine, it could also manage Flatpak, Snap and packages from the official repositories if I want to extend these functions, a bit like "Bauh" does, but from CLI;
- in addition to the aforementioned portable apps, sometimes the AppImages are also distributed in an additional .zip or .tar archive (see "voicevox", its an AppImage of several Gigabytes divided in three .7z archives);
- Libreoffice AppImage (for example) has its own system for extracting launchers and icons, being an app that includes 7 of them (Startcenter, Write, Base, Draw, Math, Calc and Impress), and also my Kdegames (40 apps) and Kdeutils (about ten) have a similar system.

Here's why I don't use a single file to manage over 1800 applications.
